### PR TITLE
[Impeller] Document ReactorGLES.

### PR DESCRIPTION
**Pull request by chinmaygarde**

### CODE REVIEW

**Comment by chinmaygarde** [comment]
The bit about multiple simultaneous reactions being safe will only be true after @jason-simmons s patch :)

File: impeller/renderer/backend/gles/reactor_gles.h [hunk 1]
```diff
@@ -15,43 +15,207 @@
...
+
+  //----------------------------------------------------------------------------
+  /// @brief      Get the OpenGL proc. table the reactor uses to manage handles.
+  ///
+  /// @return     The proc table.
+  ///
   const ProcTableGLES& GetProcTable() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Returns the OpenGL handle for reactor handle if one is
```
Comment (jason-simmons) [inline impeller/renderer/backend/gles/reactor_gles.h:142]: change to "for a reactor handle"

**Review by jason-simmons — APPROVED** [review]

File: impeller/renderer/backend/gles/reactor_gles.h [hunk 2]
```diff
@@ -15,43 +15,207 @@
 
 namespace impeller {
 
+//------------------------------------------------------------------------------
+/// @brief      The reactor attempts to make thread-safe usage of OpenGL ES
+///             easier to reason about.
+///
+///             In the other Impeller backends (like Metal and Vulkan),
+///             resources can be created, used, and consumed on any thread with
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply impeller/renderer/backend/gles/reactor_gles.h:23]: Done. Used "deleted" here and elsewhere.

File: impeller/renderer/backend/gles/reactor_gles.h [hunk 3]
```diff
@@ -15,43 +15,207 @@
...
+///             relatively few restrictions. However, OpenGL resources can only
+///             be created, used, and consumed on a thread on which an OpenGL
+///             context (or one in the same sharegroup) is  current.
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply impeller/renderer/backend/gles/reactor_gles.h:26]: Done.

File: impeller/renderer/backend/gles/reactor_gles.h [hunk 4]
```diff
@@ -15,43 +15,207 @@
...
+    ///             assumes the context is already current if true.
+    ///
     virtual bool CanReactorReactOnCurrentThreadNow(
         const ReactorGLES& reactor) const = 0;
   };
 
   using Ref = std::shared_ptr<ReactorGLES>;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Create a new reactor. There are expensive and only per
```
[the remark this answers is not in the document]
Comment (chinmaygarde) [reply impeller/renderer/backend/gles/reactor_gles.h:89]: Done.

File: impeller/renderer/backend/gles/reactor_gles.h (hunk 1 above)
Comment (chinmaygarde) [reply impeller/renderer/backend/gles/reactor_gles.h:142]: Done.

**Pull request merged**